tests/core/factory/test_checkpoint_factory.py

```diff
@@ -0,0 +1,238 @@
+from unittest.mock import Mock
+
+import pytest
+
+from great_expectations import set_context
+from great_expectations.checkpoint.checkpoint import Checkpoint
+from great_expectations.core.factory.checkpoint_factory import CheckpointFactory
+from great_expectations.data_context import AbstractDataContext
+from great_expectations.data_context.store.checkpoint_store import CheckpointStore
+from great_expectations.exceptions import DataContextError
+
+
+@pytest.fixture
+def checkpoint_dict():
+    return {
+        "name": "oss_test_checkpoint",
+        "expectation_suite_name": "oss_test_expectation_checkpoint",
+        "validations": [
+            {
+                "name": None,
+                "id": None,
+                "expectation_checkpoint_name": "taxi.demo_pass",
+                "expectation_checkpoint_ge_cloud_id": None,
+                "batch_request": None,
+            },
+        ],
+        "action_list": [
+            {
+                "action": {"class_name": "StoreValidationResultAction"},
+                "name": "store_validation_result",
+            },
+        ],
+    }
+
+
+def _assert_checkpoint_equality(actual: Checkpoint, expected: Checkpoint):
+    # Checkpoint equality is currently defined as equality of the config
+    # TODO: We should change this to a more robust comparison (instead of memory addresses)
+    actual_config = actual.config.to_json_dict()
+    expected_config = expected.config.to_json_dict()
+    assert actual_config == expected_config
+
+
+@pytest.mark.unit
+def test_checkpoint_factory_get_uses_store_get(checkpoint_dict: dict):
+    # Arrange
+    name = "test-checkpoint"
+    store = Mock(spec=CheckpointStore)
+    store.has_key.return_value = True
+    key = store.get_key.return_value
+    store.get.return_value = checkpoint_dict
+    context = Mock(spec=AbstractDataContext)
+    factory = CheckpointFactory(store=store, context=context)
+    set_context(context)
+
+    # Act
+    result = factory.get(name=name)
+
+    # Assert
+    store.get.assert_called_once_with(key=key)
+    _assert_checkpoint_equality(
+        actual=result, expected=Checkpoint(**checkpoint_dict, data_context=context)
+    )
+
+
+@pytest.mark.unit
+def test_checkpoint_factory_get_raises_error_on_missing_key(checkpoint_dict: dict):
+    # Arrange
+    name = "test-checkpoint"
+    store = Mock(spec=CheckpointStore)
+    store.has_key.return_value = False
+    store.get.return_value = checkpoint_dict
+    context = Mock(spec=AbstractDataContext)
+    factory = CheckpointFactory(store=store, context=context)
+    set_context(context)
+
+    # Act
+    with pytest.raises(
+        DataContextError, match=f"Checkpoint with name {name} was not found."
+    ):
+        factory.get(name=name)
+
+    # Assert
+    store.get.assert_not_called()
+
+
+@pytest.mark.unit
+def test_checkpoint_factory_add_uses_store_add():
```

Comment:
`add` returns the checkpoint, right? We should probably have a test asserting that.

Reply:
Will write one!